code/modules/surgery/organs/augments_internal.dm

```diff
@@ -143,6 +143,46 @@
 /obj/item/organ/cyberimp/brain/anti_stun/syndicate
 	syndicate_implant = TRUE
 
+
+/obj/item/organ/cyberimp/brain/linkedsurgery
+	name = "surgical serverlink brain implant"
+	desc = "A brain implant with a bluespace technology that lets you to perform an advanced surgery through your station research server."
+	slot = ORGAN_SLOT_SURGICAL_IMPLANT
+	actions_types = list(/datum/action/item_action/update_linkedsurgery)
+	var/list/advanced_surgeries = list()
+	var/static/datum/techweb/linked_techweb
+	var/updated
+
+/obj/item/organ/cyberimp/brain/linkedsurgery/Initialize()
+	. = ..()
+	if(isnull(linked_techweb))
+		linked_techweb = SSresearch.science_tech
+	updated = 0
+
+/obj/item/organ/cyberimp/brain/linkedsurgery/proc/update_surgery()
+	for(var/i in linked_techweb.researched_designs)
+		var/datum/design/surgery/D = SSresearch.techweb_design_by_id(i)
```

Comment:
```suggestion
	for(var/datum/design/surgery/D in linked_techweb.researched_designs)
```

Reply:
Your suggestion doesn't work because `var/D` is supposed to take a design called by `var/i`. `var/D` can't accept `linked_techweb.researched_designs` directly.